internal/storage/image.go

```diff
@@ -855,24 +815,17 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 		ctx:        ctx,
 	}
 
-	if len(registries) != 0 {
-		seenRegistries := make(map[string]bool, len(registries))
-		cleanRegistries := []string{}
-		for _, r := range registries {
-			if seenRegistries[r] {
-				continue
-			}
-			cleanRegistries = append(cleanRegistries, r)
-			seenRegistries[r] = true
-		}
-
-		is.lookup.UnqualifiedSearchRegistries = cleanRegistries
-	} else {
-		systemRegistries, err := sysregistriesv2.UnqualifiedSearchRegistries(sc)
-		if err != nil {
-			return nil, err
+	// Look up insecure registries from the registries.conf and try to mark
+	// them as insecure CIDRs.
+	insecureRegistries := []string{}
+	registries, err := sysregistriesv2.GetRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+	for _, reg := range registries {
+		if reg.Insecure {
+			insecureRegistries = append(insecureRegistries, reg.Location)
```

Comment:
- This should be `.Prefix` (the lookup key), not `.Location`
- `registries.conf` prefix values can be namespaces/repos, which this code is not ready to handle
- Using the `registries.conf` data via `IndexConfigs` changes semantics (matches by the user-specified primary name, not for each mirror individually)
- AFAICS this does not really work anyway, because CIDRs in `registries.conf`:
  - are invalid in the documented format; even if CRI-O interpreted them (and the whole implementation of CIDR matching here is completely wrong and unfixable with the Go HTTP stack, BTW), something else would happen with Podman and other consumers
  - are in practice not preserved (see `sysregistriesv2.parseLocation`), so they don’t work anyway.